### ISSUE TITLE
Build - Fails at camel-k (no --camel-k flag included, $GOPATH is set)

### ISSUE THREAD

**Issue by kahboom** [syndesisio/syndesis#6267]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ x ] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
Technically not sure whether or not it's a bug, but after a while of not running Minishift I'm trying to set things up again. I start up minishift and then run the `syndesis ui` script, which is pretty [straightforward](https://github.com/syndesisio/syndesis/blob/master/tools/bin/commands/ui) and mostly the same things I'd run manually.

The build fails during the `Building camel-k` step.

```
build.build.openshift.io/syndesis-operator-4 started
==============================================================================
Building camel-k
==============================================================================
ERROR: Can't find /Users/ryordan/.go/src/github.com/apache/camel-k. camel-k image can be build only locally (--local is implied) for camel-k builds).
ERROR: Last command exited with 1
```

The `syndesis ui` script, whether or not I run with `--nuke-everything` flag, seems to still to try to build camel-k despite there is no `--camel-k` flag. See gist here: https://gist.github.com/kahboom/13ca3256f3f0f1f33f3972a0ac9653c6

I'm on Mojave (don't judge me) and `minishift v1.33.0+ba29431`. I have also go `v.1.12.7` installed (I guess the camel-k part should work anyway, separately), this is the relative part of my `.bashrc`:

```
# GO
#export PATH=$PATH:/usr/local/opt/go/libexec/bin
export GOPATH=$HOME/.go
export PATH=$PATH:$GOPATH/bin
```

```
$ echo $GOPATH
/Users/ryordan/.go
```

Trying to delete my minishift instance now and start from scratch to see if it helps..

potentially related? https://github.com/syndesisio/syndesis/issues/4751

**Comment by zregvart** [comment]
I think this is by-product of #5620 and `camel-k` being included here:

https://github.com/syndesisio/syndesis/blob/1e9c0af08c6ed65c5a82f501317ff5c7b7b15378/tools/bin/commands/build#L305

Without specifying the list of modules to build (i.e. no `-m` parameter) build will try to build all modules in the `$MODULES`.

Not sure why are we building camel-k, we could pull the image from [docker hub](https://hub.docker.com/r/apache/camel-k).

Can someone chip in. @nicolaferraro?

**Comment by nicolaferraro** [comment]
I've not followed this specific development, but I remember that building camel k was an optional thing.. Let me have a closer look.

**Comment by nicolaferraro** [comment]
Cannot reproduce it. What's the exact command you run @kahboom ?

I've tried a bunch of `syndesis ui --something` and they work: https://gist.github.com/nicolaferraro/59637b2727d30df5bf9178f09f2f0162

there's no trace of camel k in the `ui` command. I've also tried to run `syndesis build` without args, but still the operators are not built by default:

```
[nferraro@localhost syndesis]$ syndesis build
Modules: common extension integration connector meta server s2i ui-angular ui-react test
==============================================================================
```

**Comment by kahboom** [comment]
@nicolaferraro @zregvart - Thanks for getting back to me. I was only running the following, but let me try again and put into a gist.

```
minishift start
syndesis ui --nuke-everything
```

Also tried with only  `syndesis ui` despite the script comments saying it wouldn't do anything without a flag, since that was the first command suggested to me to run.

Would you suggest I `minishift destroy` to try again? @zregvart didn't you once say bad things would happen if we keep doing that? 😅

**Comment by gashcrumb** [comment]
Do you have some weird git hook running or something?  There's no way that the `ui` subcommand intentionally runs a maven build.  It doesn't do git either, so again I've no idea how that would trigger a maven build.  It certainly doesn't here:

![image](https://user-images.githubusercontent.com/351660/62125107-21020900-b29a-11e9-8122-689f184e8273.png)

**Comment by nicolaferraro** [comment]
Yep, confirmed.

`--nuke-everything` does an install, not a build.

```
[nferraro@localhost ~]$ syndesis ui --minishift --nuke-everything
note: changing directory to /home/nferraro/git/syndesisio/syndesis/app/ui-react
Stopping the OpenShift cluster...
Cluster stopped.
Removed cache content at: '/home/nferraro/.minishift/cache'
Deleting the Minishift VM...
Minishift VM deleted.
Add-on 'admin-user' enabled
Starting minishift ....
-- Starting profile 'minishift'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.11.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.11.0' is supported ... OK
-- Checking if requested hypervisor 'kvm' is supported on this platform ... OK
-- Checking if KVM driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-kvm ... 
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present ... OK
-- Checking if Libvirt default network is active ... OK
-- Checking the ISO URL ... OK
-- Downloading OpenShift binary 'oc' version 'v3.11.0'
 53.89 MiB / 53.89 MiB [=================================================================================================================================] 100.00% 0s-- Downloading OpenShift v3.11.0 checksums ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'kvm' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    4 GB
   vCPUs :    2
   Disk size: 20 GB

   Downloading ISO 'https://github.com/minishift/minishift-centos-iso/releases/download/v1.13.0/minishift-centos7.iso'
 346.00 MiB / 346.00 MiB [===============================================================================================================================] 100.00% 0s
-- Starting Minishift VM .................. OK
-- Checking for IP address ... OK
-- Checking for nameservers ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 1% used OK
-- Writing current configuration for static assignment of IP address ... WARN
   Importing 'openshift/origin-control-plane:v3.11.0' . CACHE MISS
   Importing 'openshift/origin-docker-registry:v3.11.0' . CACHE MISS
   Importing 'openshift/origin-haproxy-router:v3.11.0'  CACHE MISS
-- OpenShift cluster will be configured with ...
   Version: v3.11.0
-- Pulling the OpenShift Container Image .................... OK
-- Copying oc binary from the OpenShift container image to VM ... OK
-- Starting OpenShift cluster ............................................................
Getting a Docker client ...
Checking if image openshift/origin-control-plane:v3.11.0 is available ...
Pulling image openshift/origin-cli:v3.11.0
E0730 10:48:32.596167    2076 helper.go:173] Reading docker config from /home/docker/.docker/config.json failed: open /home/docker/.docker/config.json: no such file or directory, will attempt to pull image docker.io/openshift/origin-cli:v3.11.0 anonymously
Image pull complete
Pulling image openshift/origin-node:v3.11.0
E0730 10:48:44.980452    2076 helper.go:173] Reading docker config from /home/docker/.docker/config.json failed: open /home/docker/.docker/config.json: no such file or directory, will attempt to pull image docker.io/openshift/origin-node:v3.11.0 anonymously
Pulled 5/6 layers, 86% complete
Pulled 6/6 layers, 100% complete
Extracting
Image pull complete
Checking type of volume mount ...
Determining server IP ...
Using public hostname IP 192.168.42.162 as the host IP
Checking if OpenShift is already running ...
Checking for supported Docker version (=>1.22) ...
Checking if insecured registry is configured properly in Docker ...
Checking if required ports are available ...
Checking if OpenShift client is configured properly ...
Checking if image openshift/origin-control-plane:v3.11.0 is available ...
Starting OpenShift using openshift/origin-control-plane:v3.11.0 ...
I0730 10:49:03.718370    2076 config.go:40] Running "create-master-config"
I0730 10:49:06.414046    2076 config.go:46] Running "create-node-config"
I0730 10:49:07.674735    2076 flags.go:30] Running "create-kubelet-flags"
I0730 10:49:08.367446    2076 run_kubelet.go:49] Running "start-kubelet"
I0730 10:49:08.749408    2076 run_self_hosted.go:181] Waiting for the kube-apiserver to be ready ...
I0730 10:50:15.814975    2076 interface.go:26] Installing "kube-proxy" ...
I0730 10:50:15.816085    2076 interface.go:26] Installing "kube-dns" ...
I0730 10:50:15.816114    2076 interface.go:26] Installing "openshift-service-cert-signer-operator" ...
I0730 10:50:15.816133    2076 interface.go:26] Installing "openshift-apiserver" ...
I0730 10:50:15.816264    2076 apply_template.go:81] Installing "openshift-apiserver"
I0730 10:50:15.816270    2076 apply_template.go:81] Installing "kube-dns"
I0730 10:50:15.816687    2076 apply_template.go:81] Installing "openshift-service-cert-signer-operator"
I0730 10:50:15.819750    2076 apply_template.go:81] Installing "kube-proxy"
I0730 10:50:21.193716    2076 interface.go:41] Finished installing "kube-proxy" "kube-dns" "openshift-service-cert-signer-operator" "openshift-apiserver"
I0730 10:52:12.398230    2076 run_self_hosted.go:242] openshift-apiserver available
I0730 10:52:12.398639    2076 interface.go:26] Installing "openshift-controller-manager" ...
I0730 10:52:12.398654    2076 apply_template.go:81] Installing "openshift-controller-manager"
I0730 10:52:15.581559    2076 interface.go:41] Finished installing "openshift-controller-manager"
Adding default OAuthClient redirect URIs ...
Adding centos-imagestreams ...
Adding registry ...
Adding sample-templates ...
Adding web-console ...
Adding router ...
Adding persistent-volumes ...
I0730 10:52:15.597821    2076 interface.go:26] Installing "centos-imagestreams" ...
I0730 10:52:15.597830    2076 interface.go:26] Installing "openshift-image-registry" ...
I0730 10:52:15.597836    2076 interface.go:26] Installing "sample-templates" ...
I0730 10:52:15.597842    2076 interface.go:26] Installing "openshift-web-console-operator" ...
I0730 10:52:15.597848    2076 interface.go:26] Installing "openshift-router" ...
I0730 10:52:15.597853    2076 interface.go:26] Installing "persistent-volumes" ...
I0730 10:52:15.598226    2076 apply_list.go:67] Installing "centos-imagestreams"
I0730 10:52:15.598550    2076 interface.go:26] Installing "sample-templates/sample pipeline" ...
I0730 10:52:15.598562    2076 interface.go:26] Installing "sample-templates/mongodb" ...
I0730 10:52:15.598568    2076 interface.go:26] Installing "sample-templates/mariadb" ...
I0730 10:52:15.598574    2076 interface.go:26] Installing "sample-templates/postgresql" ...
I0730 10:52:15.598580    2076 interface.go:26] Installing "sample-templates/django quickstart" ...
I0730 10:52:15.598586    2076 interface.go:26] Installing "sample-templates/nodejs quickstart" ...
I0730 10:52:15.598592    2076 interface.go:26] Installing "sample-templates/jenkins pipeline ephemeral" ...
I0730 10:52:15.598597    2076 interface.go:26] Installing "sample-templates/mysql" ...
I0730 10:52:15.598603    2076 interface.go:26] Installing "sample-templates/cakephp quickstart" ...
I0730 10:52:15.598617    2076 interface.go:26] Installing "sample-templates/dancer quickstart" ...
I0730 10:52:15.598623    2076 interface.go:26] Installing "sample-templates/rails quickstart" ...
I0730 10:52:15.598657    2076 apply_list.go:67] Installing "sample-templates/rails quickstart"
I0730 10:52:15.598962    2076 apply_template.go:81] Installing "openshift-web-console-operator"
I0730 10:52:15.599335    2076 apply_list.go:67] Installing "sample-templates/sample pipeline"
I0730 10:52:15.599436    2076 apply_list.go:67] Installing "sample-templates/mongodb"
I0730 10:52:15.599530    2076 apply_list.go:67] Installing "sample-templates/mariadb"
I0730 10:52:15.599650    2076 apply_list.go:67] Installing "sample-templates/postgresql"
I0730 10:52:15.599783    2076 apply_list.go:67] Installing "sample-templates/django quickstart"
I0730 10:52:15.599879    2076 apply_list.go:67] Installing "sample-templates/nodejs quickstart"
I0730 10:52:15.599970    2076 apply_list.go:67] Installing "sample-templates/jenkins pipeline ephemeral"
I0730 10:52:15.600057    2076 apply_list.go:67] Installing "sample-templates/mysql"
I0730 10:52:15.600153    2076 apply_list.go:67] Installing "sample-templates/cakephp quickstart"
I0730 10:52:15.600263    2076 apply_list.go:67] Installing "sample-templates/dancer quickstart"
I0730 10:52:27.612117    2076 interface.go:41] Finished installing "sample-templates/sample pipeline" "sample-templates/mongodb" "sample-templates/mariadb" "sample-templates/postgresql" "sample-templates/django quickstart" "sample-templates/nodejs quickstart" "sample-templates/jenkins pipeline ephemeral" "sample-templates/mysql" "sample-templates/cakephp quickstart" "sample-templates/dancer quickstart" "sample-templates/rails quickstart"
I0730 10:52:54.805478    2076 interface.go:41] Finished installing "centos-imagestreams" "openshift-image-registry" "sample-templates" "openshift-web-console-operator" "openshift-router" "persistent-volumes"
Login to server ...
Creating initial project "myproject" ...
Server Information ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.42.162:8443/console

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin


-- Applying addon 'admin-user':..
-- Exporting of OpenShift images is occuring in background process with pid 14025.
Creating project syndesis
Already on project "syndesis" on server "https://192.168.42.162:8443".
Switching to project "syndesis"
Already on project "syndesis" on server "https://192.168.42.162:8443".
Installing Syndesis CRD
cluster role "syndesis-extra-permissions" added: "developer"
Deploying Syndesis operator
Switching to project "syndesis"
Login successful.

You have access to the following projects and can switch between them with 'oc project <projectname>':

    myproject
  * syndesis

Using project "syndesis".
Waiting for syndesis-operator to be scaled to 1
Sleeping 10s ...
NAME                         READY     STATUS    RESTARTS   AGE
syndesis-operator-1-deploy   0/1       Pending   0          0s
syndesis-operator-1-deploy   0/1       Pending   0         0s
syndesis-operator-1-deploy   0/1       ContainerCreating   0         0s
syndesis-operator-1-cz6fq   0/1       Pending   0         0s
syndesis-operator-1-cz6fq   0/1       Pending   0         0s
syndesis-operator-1-cz6fq   0/1       ContainerCreating   0         0s
syndesis-operator-1-deploy   1/1       Running   0         8s
Sleeping 10s ...
Sleeping 10s ...
syndesis-operator-1-cz6fq   1/1       Running   0         21s
syndesis-operator-1-deploy   0/1       Completed   0         30s
syndesis-operator-1-deploy   0/1       Terminating   0         30s
syndesis-operator-1-deploy   0/1       Terminating   0         30s
Waiting for syndesis-server to be scaled to 1
NAME                        READY     STATUS    RESTARTS   AGE
syndesis-operator-1-cz6fq   1/1       Running   0          24s
Sleeping 10s ...
syndesis-ui-1-deploy   0/1       Pending   0         0s
syndesis-ui-1-deploy   0/1       Pending   0         0s
syndesis-ui-1-deploy   0/1       ContainerCreating   0         0s
syndesis-meta-1-deploy   0/1       Pending   0         0s
syndesis-meta-1-deploy   0/1       Pending   0         0s
syndesis-meta-1-deploy   0/1       ContainerCreating   0         0s
syndesis-db-1-deploy   0/1       Pending   0         0s
syndesis-db-1-deploy   0/1       Pending   0         0s
syndesis-db-1-deploy   0/1       ContainerCreating   0         0s
syndesis-oauthproxy-1-deploy   0/1       Pending   0         0s
syndesis-oauthproxy-1-deploy   0/1       Pending   0         0s
syndesis-oauthproxy-1-deploy   0/1       ContainerCreating   0         0s
Sleeping 10s ...
syndesis-ui-1-lf5x6   0/1       Pending   0         0s
syndesis-ui-1-lf5x6   0/1       Pending   0         0s
syndesis-meta-1-drtj8   0/1       Pending   0         0s
syndesis-server-1-deploy   0/1       Pending   0         0s
syndesis-meta-1-drtj8   0/1       Pending   0         0s
syndesis-server-1-deploy   0/1       Pending   0         0s
syndesis-ui-1-deploy   1/1       Running   0         21s
syndesis-ui-1-lf5x6   0/1       ContainerCreating   0         1s
syndesis-oauthproxy-1-dfp2b   0/1       Pending   0         0s
syndesis-db-1-rlvbd   0/2       Pending   0         0s
syndesis-oauthproxy-1-dfp2b   0/1       Pending   0         0s
syndesis-db-1-rlvbd   0/2       Pending   0         0s
syndesis-meta-1-drtj8   0/1       ContainerCreating   0         1s
syndesis-server-1-deploy   0/1       ContainerCreating   0         3s
Sleeping 10s ...
todo-1-build   0/1       Pending   0         0s
todo-1-build   0/1       Pending   0         0s
syndesis-server-1-ch9pv   0/1       Pending   0         1s
syndesis-server-1-ch9pv   0/1       Pending   0         1s
syndesis-meta-1-deploy   1/1       Running   0         25s
syndesis-prometheus-1-deploy   0/1       Pending   0         0s
syndesis-prometheus-1-deploy   0/1       Pending   0         0s
syndesis-oauthproxy-1-dfp2b   0/1       ContainerCreating   0         6s
syndesis-db-1-rlvbd   0/2       ContainerCreating   0         6s
syndesis-oauthproxy-1-deploy   1/1       Running   0         26s
syndesis-db-1-deploy   1/1       Running   0         27s
syndesis-server-1-deploy   1/1       Running   0         10s
syndesis-server-1-ch9pv   0/1       ContainerCreating   0         6s
todo-1-build   0/1       Init:0/2   0         8s
syndesis-prometheus-1-csc94   0/1       Pending   0         0s
syndesis-prometheus-1-csc94   0/1       Pending   0         0s
syndesis-prometheus-1-deploy   0/1       ContainerCreating   0         5s
syndesis-prometheus-1-deploy   1/1       Running   0         6s
syndesis-prometheus-1-csc94   0/1       ContainerCreating   0         3s
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
syndesis-ui-1-lf5x6   0/1       Running   0         55s
syndesis-ui-1-lf5x6   1/1       Running   0         58s
syndesis-ui-1-deploy   0/1       Completed   0         1m
syndesis-ui-1-deploy   0/1       Terminating   0         1m
syndesis-ui-1-deploy   0/1       Terminating   0         1m
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
syndesis-meta-1-drtj8   0/1       Running   0         2m
Sleeping 10s ...
todo-1-build   0/1       Init:0/2   0         2m
Sleeping 10s ...
todo-1-build   0/1       Init:1/2   0         2m
syndesis-meta-1-drtj8   1/1       Running   0         2m
syndesis-meta-1-deploy   0/1       Completed   0         2m
todo-1-build   0/1       PodInitializing   0         2m
syndesis-meta-1-deploy   0/1       Terminating   0         3m
syndesis-meta-1-deploy   0/1       Terminating   0         3m
Sleeping 10s ...
todo-1-build   1/1       Running   0         3m
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
syndesis-oauthproxy-1-dfp2b   0/1       Running   0         3m
Sleeping 10s ...
syndesis-oauthproxy-1-dfp2b   1/1       Running   0         4m
syndesis-oauthproxy-1-deploy   0/1       Completed   0         4m
syndesis-oauthproxy-1-deploy   0/1       Terminating   0         4m
syndesis-oauthproxy-1-deploy   0/1       Terminating   0         4m
Sleeping 10s ...
syndesis-server-1-ch9pv   0/1       Running   0         4m
Sleeping 10s ...
syndesis-prometheus-1-csc94   0/1       Running   0         4m
Sleeping 10s ...
syndesis-db-1-rlvbd   0/2       Running   0         4m
syndesis-db-1-rlvbd   1/2       Running   0         5m
Sleeping 10s ...
syndesis-server-1-ch9pv   0/1       Error     0         5m
Sleeping 10s ...
syndesis-server-1-ch9pv   0/1       Running   1         5m
syndesis-prometheus-1-csc94   1/1       Running   0         5m
syndesis-prometheus-1-deploy   0/1       Completed   0         5m
syndesis-prometheus-1-deploy   0/1       Terminating   0         5m
syndesis-prometheus-1-deploy   0/1       Terminating   0         5m
Sleeping 10s ...
syndesis-db-1-rlvbd   2/2       Running   0         5m
syndesis-db-1-deploy   0/1       Completed   0         6m
syndesis-db-1-deploy   0/1       Terminating   0         6m
syndesis-db-1-deploy   0/1       Terminating   0         6m
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
todo-1-build   0/1       Completed   0         6m
todo-1-deploy   0/1       Pending   0         0s
todo-1-deploy   0/1       Pending   0         0s
todo-1-deploy   0/1       ContainerCreating   0         0s
Sleeping 10s ...
todo-1-hb4pt   0/1       Pending   0         0s
todo-1-hb4pt   0/1       Pending   0         0s
todo-1-deploy   1/1       Running   0         2s
todo-1-hb4pt   0/1       ContainerCreating   0         0s
todo-1-hb4pt   1/1       Running   0         3s
todo-1-deploy   0/1       Completed   0         6s
todo-1-deploy   0/1       Terminating   0         7s
todo-1-deploy   0/1       Terminating   0         7s
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
syndesis-server-1-ch9pv   1/1       Running   1         6m
syndesis-server-1-deploy   0/1       Completed   0         6m
syndesis-server-1-deploy   0/1       Terminating   0         6m
syndesis-server-1-deploy   0/1       Terminating   0         6m
Waiting for syndesis-ui to be scaled to 1
Waiting for syndesis-meta to be scaled to 1
Patching imagestream syndesis-server
imagestream.image.openshift.io/syndesis-server patched
Patching imagestream syndesis-ui
imagestream.image.openshift.io/syndesis-ui patched
Patching imagestream syndesis-meta
imagestream.image.openshift.io/syndesis-meta patched
Patching imagestream syndesis-s2i
imagestream.image.openshift.io/syndesis-s2i patched
Patching imagestream syndesis-operator
imagestream.image.openshift.io/syndesis-operator patched
Done!
~
note: changing directory back to /home/nferraro
```

**Comment by kahboom** [comment]
Closing, I am able to avoid it when I use the `--minishift` flag, I didn't even realize it was there (overlooked it). https://gist.github.com/kahboom/7acfda82bcd1a777eb44ca2a25633887

Having said that, I do still need to be able to build as the point was to get my environment properly set up, and that's not just for install. Will come back to this on the syndesis-ui chat.
